src/tests/string/memchr_test.c

```diff
@@ -7,11 +7,9 @@ TEST_CASE("memchr") {
     /* Source string */
     char src[] = "ABCCDEF";
 
-    char *ptr = memchr(src, 'C', sizeof(src));
     /* Checking that a character is found */
-    test_assert_equal(ptr, src + 2);
+    test_assert_equal(memchr(src, 'C', sizeof(src)), src + 2);
 
-    ptr = memchr(src, 1, sizeof(src));
     /* Checking that a character is not found */
```

Comment:
Could you please split TEST_CASE to 2 case 
    /* Checking that a character is found */
    and 
        /* Checking that a character is not found */